lib/paper_trail/attribute_serializers/cast_attribute_serializer.rb

```diff
@@ -34,8 +34,13 @@ def initialize(klass)
         end
 
         def serialize(attr, val)
-          val = defined_enums[attr][val] if defined_enums[attr]
-          @klass.type_for_attribute(attr).type_cast_for_database(val)
+          castable_val = val
+          if defined_enums[attr]
+            # `attr` is an enum. Find the number that corresponds to `val`. If `val` is
+            # a number already, there won't be a corresponding entry, just use `val`.
+            castable_val = defined_enums[attr][val] || val
```

Comment:
I think we do need the disjunction, otherwise `castable_val` could be incorrectly set to `nil`.